std/random.d

```diff
@@ -2634,6 +2634,104 @@ do
     assert(i == 0);
 }
 
+/+ @nogc bool array designed for RandomCover.
+- constructed with an invariable length
+- small length means 0 alloc and bit field (if up to 32 and 64 elements to cover)
+- bigger length means non-GC heap allocation(s) and dealloc. +/
+private struct RandomCoverChoices
+{
+    private static immutable emsg = "No memory available to store random cover choices";
+    private void* buffer;
+    private immutable size_t _length;
+    private immutable bool hasPackedBits;
+
+    this(this) nothrow @nogc @trusted
+    {
+        import core.stdc.stdlib : malloc;
```

Comment:
[`core.memory : pureMalloc`, `pureCalloc`, `pureFree`](https://dlang.org/phobos/core_memory.html#.pureMalloc)?

Reply:
Although since you `assert(0)` if they fail, it would be justifiable to call "locally `pure`ified" `malloc`/etc. directly.

Reply:
i've moved to `onOutOfMemoryErrorNoGC` which is not pure so this is not an option anymore.
Now there's a dilemma : _pure allocs and asserts_ or _static exceptions and impure_

Reply:
`onOutOfMemoryError` is `@nogc nothrow pure`, so why not use it instead of `onOutOfMemoryErrorNoGC`?